pkg/util/deployment/deployment.go

```diff
@@ -140,11 +140,31 @@ func ListPods(deployment *extensions.Deployment, getPodList podListFunc) (*api.P
 	return getPodList(namespace, options)
 }
 
+// equalIgnoreHash returns true if two given podTemplateSpec are equal, ignoring the diff in value of Labels[pod-template-hash]
+// We ignore pod-template-hash because the hash result would be different upon podTemplateSpec API changes
```

Comment:
Given an example of an API change that would caused this (e.g. the addition of a new defaulted field will cause the hash code to change even when it is unspecified due to the defaulting.)